Userland/Libraries/LibARM64/Instruction.cpp

```diff
@@ -0,0 +1,2539 @@
+/*
+ * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/StringBuilder.h>
+#include <LibARM64/Instruction.h>
+#include <LibARM64/Interpreter.h>
+
+#if defined(AK_COMPILER_GCC)
+#    pragma GCC optimize("O3")
+#endif
+
+namespace ARM64 {
+
+InstructionDescriptor s_table[3][256];
+InstructionDescriptor s_0f_table[3][256];
+InstructionDescriptor s_sse_table_np[256];
+InstructionDescriptor s_sse_table_66[256];
+InstructionDescriptor s_sse_table_f3[256];
+InstructionDescriptor s_sse_table_f2[256];
+
+static bool opcode_has_register_index(u8 op)
+{
+    if (op >= 0x40 && op <= 0x5F)
+        return true;
+    if (op >= 0x90 && op <= 0x97)
+        return true;
+    if (op >= 0xB0 && op <= 0xBF)
+        return true;
+    return false;
+}
+
+static void build_in_table(InstructionDescriptor* table, u8 op, char const* mnemonic, InstructionFormat format, InstructionHandler handler, IsLockPrefixAllowed lock_prefix_allowed)
+{
+    InstructionDescriptor& d = table[op];
+
+    d.handler = handler;
+    d.mnemonic = mnemonic;
+    d.format = format;
+    d.lock_prefix_allowed = lock_prefix_allowed;
+
+    if ((format > __BeginFormatsWithRMByte && format < __EndFormatsWithRMByte) || format == MultibyteWithSlash)
+        d.has_rm = true;
+    else
+        d.opcode_has_register_index = opcode_has_register_index(op);
+
+    switch (format) {
+    case OP_RM8_imm8:
+    case OP_RM16_imm8:
```

Comment:
This file is clearly unrelated to aarch64. Let's not just copy paste Libx86 files and call them "LibARM64". SSE is not a thing in aarch64. these instruction formats are not valid for aarch64.